packages/react-components/src/components/ResponsiveVerticalGallery.tsx

```diff
@@ -0,0 +1,113 @@
+  /** Video tiles to be rendered in the Vertical Gallery */
+  children: React.ReactNode;
+  /** Styles for the Children space container */
+  containerStyles: IStyle;
+  /** Styles for the VerticalGallery component */
+  verticalGalleryStyles: VerticalGalleryStyles;
+  /** Height of the gap in between the video tiles */
+  gapHeightRem: number;
+  /** Height of the control bar for navigating pages */
+  controlBarHeightRem?: number;
+}
+
+/**
+ * Responsive container for the VerticalGallery Component. Performs calculations for number of children
+ * for the VerticalGallery
+ * @param props
+ *
+ * @beta
+ */
+export const ResponsiveVerticalGallery = (props: ResponsiveVerticalGalleryProps): JSX.Element => {
+  const { children, containerStyles, verticalGalleryStyles, gapHeightRem, controlBarHeightRem } = props;
+  const containerRef = useRef<HTMLDivElement>(null);
+  const containerHeight = _useContainerHeight(containerRef);
+
+  const topPadding = containerRef.current ? parseFloat(getComputedStyle(containerRef.current).paddingTop) : 0;
+  const bottomPadding = containerRef.current ? parseFloat(getComputedStyle(containerRef.current).paddingBottom) : 0;
+
+  const childrenPerPage = calculateChildrenPerPage({
+    numberOfChildren: React.Children.count(children),
+    containerHeight: (containerHeight ?? 0) - topPadding - bottomPadding,
+    gapHeightRem,
+    controlBarHeight: controlBarHeightRem ?? 2
+  });
+  return (
+    <div ref={containerRef} className={mergeStyles(containerStyles)}>
+      <VerticalGallery childrenPerPage={childrenPerPage} styles={verticalGalleryStyles}>
+        {children}
+      </VerticalGallery>
+    </div>
+  );
+};
+
+/**
+ * Helper function to find the number of children for the VerticalGallery on each page.
+ */
+const calculateChildrenPerPage = (args: {
+  numberOfChildren: number;
+  containerHeight: number;
+  gapHeightRem: number;
+  controlBarHeight: number;
+}): number => {
+  const { numberOfChildren, containerHeight, gapHeightRem, controlBarHeight } = args;
+
+  const childMinHeightPx = _convertRemToPx(LARGE_VERTICAL_GALLERY_TILE_SIZE_REM.minHeight);
+  const gapHeightPx = _convertRemToPx(gapHeightRem);
+  const controlBarHeightPx = _convertRemToPx(controlBarHeight);
+
+  /** First check how many children can fit in containerHeight.
+   *
+   *   _________________
+   *   |                |
+   *   |                |
+   *   |________________|
+   *    _________________
+   *   |                |
+   *   |                |
+   *   |________________|
+   *
+   *      <   n/m   >
+   *
+   * number of children = container height - (2* gap height + button height) / childMinHeight
+   *
+   * we want to find the maximum number of children at the smallest size we can fit in the gallery and then resize them
+   * to fill in the space as much as possible
+   */
+
+  const numberOfChildrenInContainer = Math.floor(
+    (containerHeight - (2 * gapHeightPx + controlBarHeightPx)) / childMinHeightPx
+  );
```

Comment:
This first calculation should be to check if all children will fit into the container height without needing to paginate. So you should not account for the control bar height and the gaps for it. Also you must take into account the gaps between the children like so:
containerHeight = n * childMinHeightPx+ (n - 1) * gapHeightPx -> isolate n and take the floor -> Math.floor((containerWidth + gapWidth) / (childWidth + gapWidth))